pynbody/snapshot/tipsy.py

```diff
@@ -1182,6 +1183,31 @@ def __init__(self, filename, sort=True, paramfile=None):
                                                'f8', 'f8', 'f8',
                                                'f8', 'f8', 'f8','f8')})
             molecH = True
+            # Unfortunately molecularH with small iOrders has the same as
+            # no moleculuarH with big iOrders.  Attempt to distinguish here
+            if(iSize == file_structure.itemsize):
+                if(self._byteswap):
+                    testread = np.fromstring(
+                        f.read(iSize), dtype=file_structure).byteswap()
+                else:
+                    testread = np.fromstring(f.read(iSize), dtype=file_structure)
+                # All star iorders are greater than any gas iorder
+                # so this indicates a bad format. (N.B. there is the
+                # possibility of a false negative)
+                if(testread['iord'][0] < testread['iorderGas'][0]): 
+                    file_structure = np.dtype({'names': ("iord", "iorderGas",
+                                             "tform",
+                                             "x", "y", "z",
+                                             "vx", "vy", "vz",
+                                             "massform", "rhoform", "tempform"),
+                                       'formats': ('i8', 'i8', 'f8',
+                                                   'f8', 'f8', 'f8',
+                                                   'f8', 'f8', 'f8',
+                                                   'f8', 'f8', 'f8')})
+                    f.seek(4)
+                    print "Using 64 bit iOrders"
```

Comment:
Could you change this to `logger.info("Using 64 bit iOrders")` for compatibility with the logging elsewhere? 

Otherwise, I can merge this in! Thanks